pkg/prediction/percentile/prediction.go

```diff
@@ -276,14 +286,31 @@ func (p *percentilePrediction) Run(stopCh <-chan struct{}) {
 			// todo: Do not block this management go routine here to do some time consuming operation.
 			// We just init the signal and setting the status
 			// we start the real time model updating directly. but there is a window time for each metricNamer in the algorithm config to ready status
-			if err := p.init(qc.MetricNamer); err != nil {
-				klog.ErrorS(err, "Failed to init percentilePrediction.")
+			c := p.a.GetConfig(QueryExpr)
+			if c == nil {
+				c = &defaultInternalConfig
+			}
+
+			var initError error
+			switch c.initMode {
+			case config.ModelInitModeLazyTraining:
```

Comment:
lazy training mode is not from history?

Reply:
> lazy training mode is not from history?

yes, it means accumulating data from real time data source, then when the data length is enough, it can do predict. i have not thought a better naming